experimental/oak_baremetal_channel/SPEC.MD

```diff
@@ -0,0 +1,121 @@
+# Oak Baremetal Communication Protocol Specification
+
+This document serves as a specification of the protocol used for communication
+between the Untrusted Loader and Trusted Runtime. The Trusted Runtime should be
+able to communicate with any implementation of the Untrusted Loader that
+implements this specification.
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
+"SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be
+interpreted as described in [RFC 2119](https://www.rfc-editor.org/rfc/rfc2119).
+
+## Method
+
+Methods are exposed by the Trusted Runtime. They MAY be remotely invoked by the
+Unrusted Loader. Each method has clearly defined parameters and return types.
+The Oak IDL is used to define the methods exposed by the Trusted Runtime.
+
+## Invocation
+
+An invocations of a Method. Each invocation consists out of a Request Message
+followed by a Response Message.
+
+### Issuing an invocation
+
+The Untrusted Loader MAY issue an invocation by sending a Request Message.
+
+The Trusted Runtime MUST NOT send Request Message.
+
+### Handling an invocation
+
+The Trusted Runtime MUST respond to each Request Message with a single Response
+Message. It SHOULD respond to Request Messages in first-in-first-out order, but
+MAY prioritize some requests. The Trusted Runtime MAY issue Response Messages
+before the Request Message has been received in its entirety. Sending a Response
+Messages concludes the invocation.
+
+The Untrusted Loader MUST NOT send Response Messages.
+
+## Messages
+
+A message forms one half of an Invocation.
+
+### Request Message Definition
+
+A byte encoded request message MUST consist of the following seqeuential fields:
+
+- length, u32, little endian: length of the entire message, including this field
+- invocation_id, u32, little endian: Identifies the invocation the request
+  initiates
+- method_id, u32, little endian: identifies the method that is being invoked
+- padding, 4 bytes: reserved to maintain 8 byte alignment. MUST be unset and
+  ignored
+- body, variable length byte array: the method’s arguments, encoded with
```

Comment:
nit: "encoded with flatbuffer" feels a bit weird. I am not 100% sure about the correct terminology, but would have expected something more like "encoded as FlatBuffers"

Reply:
Fair, moved the mention of the encoding protocol up to method section, and replaced this. :)